docs/proposals/extensions.md

```diff
@@ -0,0 +1,88 @@
+## Proposal for extending an OpenCompose file with another OpenCompose file
+
+##### Scenario:
+
+Let's say, we have a sample OpenCompose file defining a Wordpress application -
+
+_wordpress-opencompose.yml_ -
+```yaml
+version: '0.1-dev'
+
+services:
+- name: database
+  containers:
+  - image: mariadb:10
+    env:
+    - name: MYSQL_ROOT_PASSWORD
+      value: rootpasswd
+    - name: MYSQL_DATABASE
+      value: wordpress
+    - name: MYSQL_USER
+      value: wordpress
+    - name: MYSQL_PASSWORD
+      value: wordpress
+    ports:
+    - port: 3306
+
+- name: web
+  containers:
+  - image: wordpress:4
+    env:
+    - name: WORDPRESS_DB_HOST
+      value: database:3306
+    - name: WORDPRESS_DB_PASSWORD
+      value: wordpress
+    - name: WORDPRESS_DB_USER
+      value: wordpress
+    - name: WORDPRESS_DB_NAME
+      value: wordpress
+    ports:
+    - port: 80
+      type: external
+```
+
+This file works fine for a developer, but since there is no persistent volumes defined here, we need to alter this for production use cases; but changing the original file disrupts the developer workflow.
+
+So, the production folks create another OpenCompose file which will extend _wordpress-opencompose.yml_ which will look like -
+
+_wordpress.extension.yml_
+
+```yaml
+version: '0.1-dev'
+extends: wordpress-opencompose.yml
+
+services:
+- name: database
+  env:
```

Comment:
yes, we will have to do that.
One thing that we might consider is making it optional for services with only one container. In this case, container will have the same name as the services.